src/DrDotnet.Profilers/profilers/cpu_hotpath_profiler.rs

```diff
@@ -176,116 +41,175 @@ impl Profiler for CpuHotpathProfiler {
                     type_: ParameterType::INT.into(),
                     value: "40".to_owned(),
                     ..std::default::Default::default()
+                },
+                ProfilerParameter { 
+                    name: "Filter Suspended Threads".to_owned(),
+                    key: "filter_suspended_threads".to_owned(),
+                    description: "If set, the profiler will attempt to detect suspended thread and filter them out from the analysis to only focus on working threads, hereby improving accuracy on actual CPU hotpaths".to_owned(),
+                    type_: ParameterType::BOOLEAN.into(),
+                    value: "true".to_owned(),
+                    ..std::default::Default::default()
                 }
             ],
             ..std::default::Default::default()
         }
     }
 }
 
+#[derive(Default)]
+pub struct CpuHotpathStackSnapshotCallbackReceiver {
+    method_ids: Vec::<FunctionID>,
+    hasher: GxHasher
+}
+
+impl StackSnapshotCallbackReceiver for CpuHotpathStackSnapshotCallbackReceiver {
+    type AssociatedType = Self;
+
+    fn callback(&mut self, method_id: FunctionID, instruction_pointer: usize, _frame_info: usize, _context: &[u8]) {
+        // Filter out unmanaged stack frames
+        if method_id == 0 {
+            return;
+        }
+        self.method_ids.push(method_id);
+        // Detect suspended threads appart from actual working threads
+        // Inspired from: https://www.usenix.org/legacy/publications/library/proceedings/coots99/full_papers/liang/liang_html/node10.html
+        // Not sure which is the best approach between utilizing the instruction pointer or the context
+        self.hasher.write_usize(instruction_pointer);
+        //self.hasher.write(context);
```

Comment:
from my understanding context looks like the appropriate structure to hash (but I honestly don't really get instruction pointer). 
If we go for `instruction_pointer` we don't need the `COR_PRF_SNAPSHOT_CONTEXT` flag to `ICorProfilerInfo2::DoStackSnapshot` and `threads_by_context_hash` can be renamed `threads_by_instruction_pointer_hash`

Reply:
I've ran a test in production using instruction pointer, I'll run another test using the context and see how it differs

Reply:
Both seem to yield comparable results on production

Reply:
I'll stick with instruction_pointer which is probably more lightweight